src/main/scala/is/hail/table/Table.scala

```diff
@@ -158,6 +158,26 @@ object Table {
         RVD(signature, rdd2))
     ))
   }
+
+  def sameWithinTolerance(t: Type, l: Array[Row], r: Array[Row], tolerance: Double): Boolean = {
+    val used = new Array[Boolean](r.length)
+    var i = 0
+    while (i < l.length) {
+      val li = l(i)
+      var matched = false
+      var j = 0
+      while (!matched && j < l.length && !used(j)) {
+        matched = t.valuesSimilar(li, r(j), tolerance)
```

Comment:
I don't understand why you are doing a product comparison here. Shouldn't it be a zip operation?

Reply:
IIUC, the array of rows here are all the values that had the same key in the left table and the right table, respectively.

So I don't know ahead of time the right ordering of the values on the left and the values on the right such that they're similar. For example, if the left had 1.0 and 2.0 and right had 1.9 and 0.9, and the tolerance was 1.0, I'd want to match the 1.0 with the 0.9 and the and 1.9 with the 2.0. But to find that particular set of pairs, I, in general, need to check all possible set of pairs.